src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,111 @@
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product productFromDb = productService.findById(id);
+        productFromDb.setTitle(productRequestDto.getTitle());
+        productFromDb.setPrice(productRequestDto.getPrice());
+        return productMapper.mapToDto(productService.save(productFromDb));
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Get all products.")
+    public List<ProductResponseDto> findAll(@RequestParam(required = false)
+                                            @ApiParam(value = "Optional parameter.")
+                                            BigDecimal priceFrom,
+                                            @RequestParam(required = false)
+                                            @ApiParam(value = "Optional parameter.")
+                                            BigDecimal priceTo,
+                                            @RequestParam(defaultValue = "20")
+                                            @ApiParam(value = "Default value - 20.")
+                                            Integer size,
+                                            @RequestParam(defaultValue = "0")
+                                            @ApiParam(value = "Default value - 0.")
+                                            Integer page,
+                                            @RequestParam(defaultValue = "id")
+                                            @ApiParam(value = "Sort by default - by ID.")
+                                            String sortBy) {
+        List<Sort.Order> orders = new ArrayList<>();
+        if (sortBy.contains(":")) {
+            String[] sortingFields = sortBy.split(";");
+            for (String field : sortingFields) {
+                Sort.Order order;
+                if (field.contains(":")) {
+                    String[] fieldsAndDirections = field.split(":");
+                    order = new Sort.Order(Sort.Direction.valueOf(fieldsAndDirections[1]),
+                            fieldsAndDirections[0]);
+                } else {
+                    order = new Sort.Order(Sort.Direction.DESC, field);
+                }
+                orders.add(order);
+            }
+        } else {
+            Sort.Order order = new Sort.Order(Sort.Direction.DESC, sortBy);
+            orders.add(order);
+        }
+        Sort sort = Sort.by(orders);
+        PageRequest pageRequest = PageRequest.of(page, size, sort);
+        return priceFrom != null || priceTo != null
+                ? productService.findAll(priceFrom, priceTo,
+                                pageRequest).stream()
+                        .map(productMapper::mapToDto)
+                        .collect(Collectors.toList())
+                : productService.findAll(pageRequest).stream()
+                        .map(productMapper::mapToDto)
+                        .collect(Collectors.toList());
```

Comment:
It is too much logic for the controller. Isn't it a violation of SRP?)

Reply:
It is also better to refactor it a bit and move something into separate private methods and constants